text/0000-export-to-oci.md

```diff
@@ -0,0 +1,261 @@
+- A [Lifecycle](https://buildpacks.io/docs/concepts/components/lifecycle/) orchestrates Buildpacks execution, then assembles the resulting artifacts into a final app image.
+- A Daemon is a service, popularized by Docker, for downloading container images, and executing and managing containers from those images.
+- A Registry is a long-running service used for storing and retrieving container images.
+- A digest reference refers to a [content addressable](https://en.wikipedia.org/wiki/Content-addressable_storage) identifier of form <registry>/<repo>@<digest> which locates an image manifest in an [OCI Distribution Specification](https://github.com/opencontainers/distribution-spec/blob/master/spec.md) compliant registry.
+- A Image Manifest provides a configuration and set of layers for a single container image for a specific architecture and operating system.
+- An [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) is the directory structure for OCI content-addressable blobs and location-addressable references.
+
+# Motivation
+[motivation]: #motivation
+
+Implementing this new feature will help us to solve the problem of loosing information when the image is saved into the Daemon keeping the image on disk along with the metadata it can be used as input for other tools to offer more capabilities to the end users.
+
+This feature will help to unblock uses cases like
+- OCI annotations. See [RFC](https://github.com/buildpacks/rfcs/pull/196)
+- Cosign integration. See [RFC](https://github.com/buildpacks/rfcs/pull/195)
+
+# What it is
+[what-it-is]: #what-it-is
+
+Currently the *Exporter*  writes either in an OCI image registry or a docker daemon, the idea is to add the capability to write into disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format when the new flag `-layout` or the default environment variable `CNB_LAYOUT_DIR` is set.
+
+Let's see some examples for the propose behavior
+
+## Examples
+
+For each case, I will present two ways of invoking the new feature:
+
+- Using the environment Variable
+- Using the new flag
+
+For both ways the expected output is the same
+
+### Exporting to Daemon with launch cache enabled
+
+```=shell
+> export CNB_LAYOUT=oci
+> /cnb/lifecycle/exporter -daemon -launch-cache /launch-cache my-app-image
+```
+
+Or
+
+```=shell
+> /cnb/lifecycle/exporter -daemon -launch-cache /launch-cache -layout /oci my-app-image
+```
+
+The expected output is the `my-app-image` exported in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format into the `/launch-cache/my-app-image/` folder.
+
+```=shell
+> cd /launch-cache
+> tree .
+.
+└── launch-cache/
+    ├── committed/
+    │   ├── io.buildpacks.lifecycle.cache.metadata
+    │   ├── sha256:65d9067f915e01...tar
+    │   ├── sha256:6905011516dcf4...tar
+    │   └── sha256:83d85471d9f8a3...tar
+    ├── staging
+    └── my-app-image/
```

Comment:
This could still be a child of `/oci`, no? Since we are using symlinks, I would think we don't really have to care.

Reply:
Yes, my doubt is:
-   if I am referencing files that are saved in the launch cache (mounted in a volume by tools like Pack) then the final exported images MUST be in the same folder to guarantee we have access to those files